tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenAssertIndexNotExists.cs

```diff
@@ -0,0 +1,80 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using MongoDB.Bson;
+
+namespace MongoDB.Driver.Tests.JsonDrivenTests
+{
+    public class JsonDrivenAssertIndexNotExists : JsonDrivenTestRunnerTest
+    {
+        // private fields
+        private string _collectionName;
+        private string _databaseName;
+        private string _indexName;
+
+        // public constructors
+        public JsonDrivenAssertIndexNotExists(IJsonDrivenTestRunner testRunner, Dictionary<string, object> objectMap)
+            : base(testRunner, objectMap)
+        {
+        }
+
+        // public methods
+        public override void Act(CancellationToken cancellationToken)
+        {
+            // do nothing
+        }
+
+        public override Task ActAsync(CancellationToken cancellationToken)
+        {
+            // do nothing
+            return Task.FromResult(true);
+        }
+
+        public override void Assert()
+        {
+            using (var client = DriverTestConfiguration.CreateDisposableClient())
+            {
+                var indexesCursor = client.GetDatabase(_databaseName).GetCollection<BsonDocument>(_collectionName).Indexes.List();
+                var indexes = indexesCursor.ToList();
+                var indexFound = indexes.Any(x => x["name"] == _indexName);
+                indexFound.Should().BeFalse();
+            }
+        }
```

Comment:
Simplify Assert method to:

    public override void Assert()
    {
        var client = DriverTestConfiguration.Client;
        var indexes = client.GetDatabase(_databaseName).GetCollection<BsonDocument>(_collectionName).Indexes.List().ToList();
        var indexNames = indexes.Select(i => i["name"].AsString);
        indexNames.Should().NotContain(_indexName);
    }

Reply:
Done.